Задание спринта 14, Add bookings

NinaShch opened the pull request.

NinaShch committed:
Задание спринта 14

Umka7389 commented on src/main/java/ru/practicum/shareit/ShareItApp.java:
```diff
@@ -2,12 +2,13 @@
+import org.springframework.data.jpa.repository.config.EnableJpaRepositories;
 
 @SpringBootApplication
+@EnableJpaRepositories
```
Это вроде бы лишняя аннотация, у вас репозитории находятся в том же пакете

Umka7389 commented on src/main/java/ru/practicum/shareit/booking/BookingGetState.java:
```diff
@@ -0,0 +1,3 @@
+package ru.practicum.shareit.booking;
+
+enum BookingGetState { ALL, CURRENT, PAST, FUTURE, WAITING, REJECTED }
```
Мб лучше просто BookingState?

Umka7389 commented on src/main/java/ru/practicum/shareit/booking/BookingRepository.java:
```diff
@@ -0,0 +1,52 @@
+import java.time.LocalDateTime;
+import java.util.List;
+
+@Repository
```
Если наследуетесь от JpaRepository можно обойтись без этой аннотации

Umka7389 commented on src/main/java/ru/practicum/shareit/item/storage/InMemoryItemStorage.java:
```diff
@@ -0,0 +1,62 @@
+package ru.practicum.shareit.item.storage;
```
InMemory вам наверное уже не нужный

Umka7389 commented on pom.xml:
```diff
@@ -21,38 +21,40 @@
 		<dependency>
-			<groupId>org.springframework.boot</groupId>
-			<artifactId>spring-boot-starter-web</artifactId>
+			<groupId>org.projectlombok</groupId>
```
Вы, наверное, должны мержить прошлые реквесты. Вы присылаете новый PR в котором изменения прошлых версий тоже содержатся, приходится все смотреть заново, обсудите с наставником

Umka7389 commented on src/main/java/ru/practicum/shareit/booking/BookingController.java:
```diff
@@ -1,12 +1,78 @@
+    private final BookingService bookingService;
+
+    @PostMapping
+    public Booking createBooking(
```
На фронт возвращается обычно Dto сущность иначе зачем она?

Umka7389 commented on src/main/java/ru/practicum/shareit/booking/dto/ExtremumBookingDto.java:
```diff
@@ -0,0 +1,14 @@
+
+@Data
+@AllArgsConstructor
+public class ExtremumBookingDto {
```
Непонятный нейминг. Мб лучше сделать вложенным классом в ItemDto, а не отдельным?

Umka7389 commented on src/main/java/ru/practicum/shareit/booking/BookingMapper.java:
```diff
@@ -0,0 +1,26 @@
+
+    public static ExtremumBookingDto toExtremumBookingDto(Booking booking) {
+        return new ExtremumBookingDto(booking.getId(), booking.getBooker().getId());
+    }
```
Обратный маппинг было бы тоже неплохо добавить

Umka7389 commented on src/main/java/ru/practicum/shareit/booking/BookingRepository.java:
```diff
@@ -0,0 +1,52 @@
+@Repository
+public interface BookingRepository extends JpaRepository<Booking, Long> {
+
+    @SortComparator(BookingDateComparator.class)
```
Интересное решение. Как вариант можно в сервисе сделать 
Sort newestFirst = Sort.by(Sort.Direction.DESC, "start"); 
и в метод репозитория передавать вторым аргументом. Хотя этот способ тоже имеет место быть, можешь оставить на свое усмотрение

Umka7389 reviewed (commented):
Хорошая работа, молодец! Замечания по коду

NinaShch committed:
Задание спринта 14, исправления
То что прошлый ПР не смерджила мой косяк, раньше всегда мерджила перед новым ПР, в этот раз просто забыла

Umka7389 reviewed (commented):
Отличная работа, молодец!
